### ISSUE TITLE
Some objects don't work when the DAW is not sending audio blocks in

### ISSUE THREAD

**Issue by timothyschoen** [timothyschoen/PlugData#157]
Some DAWs stop sending/requesting audio blocks when audio is paused/stopped, or even when there is no audio on the track going in. PlugData requires that audio is running because that's what drives pure-data's scheduler. No audio means no time-based events.

I wrote a fix for this that starts a backup thread whenever audio stops, but this is still too unstable to use right now.

Objects that are affected include time-based things like [metro] and [rec], but also objects like [netreceive] that need to check for updates constantly.

**Comment by Yzrsah** [comment]
@timothyschoen It might be neat if the power/mute button had a theme color to remind you when the DAW is not sending any audio blocks in. The power button's highlight could turn I.e. Purple to show when it's not getting any air :) Alternatively the icon could change or some other indicator could change.

**Comment by timothyschoen** [comment]
> @timothyschoen It might be neat if the power/mute button had a theme color to remind you when the DAW is not sending any audio blocks in. The power button's highlight could turn I.e. Purple to show when it's not getting any air :) Alternatively the icon could change or some other indicator could change.

That's a great idea, coming up!

**Comment by timothyschoen** [comment]
<img width="366" alt="Screenshot 2023-01-28 at 23 02 58" src="https://user-images.githubusercontent.com/44585538/215293087-3f8f1a9c-1b7c-48cd-8a94-a9c0c8529812.png">

Done!

**Comment by alcomposer** [comment]
Be good to show the current state in text as hover for power button.

Also, I wonder if power button is the correct icon for this. Should it not be mute?

**Comment by tomara-x** [comment]
yeah, agree with the text, color alone doesn't communicate what's happening

it does enable/disable all dsp, so i think the icon is appropriate

**Comment by alcomposer** [comment]
Maybe we need two volume meters to make it clearer? In and out.

Or do it like the midi message light?

When I see a power button light change colour, I don't necessarily think good things are happening.

**Comment by tomara-x** [comment]
maybe do it the vanilla way (the word DSP with a check mark next to it)
![image](https://user-images.githubusercontent.com/86204514/216874990-7230e619-a83b-45c2-8bce-5d24181bdcec.png)
and the word itself can change color? idk
